science/nds2-client/Portfile

```diff
@@ -13,117 +16,169 @@ long_description \
+test.target        -R '.*'
 
-            depends_build-append    port:swig-python
-            depends_lib-append      port:${p} port:py${s}-numpy
-            configure.args-strsed   s/--disable-swig-python/--enable-swig-python/
-            destroot.args-append    pythondir="${d}" pyexecdir="${d}"
+use_parallel_build yes
 
-        }
-    }]
-}
+livecheck.type    regex
+livecheck.url     ${master_sites}
+livecheck.regex   {nds2-client-(\d+(?:\.\d+)*).tar.gz}
 
-variant swig_octave description "Enable SWIG Octave interface" {
+#========================================================================
+# Create subports for Java
+#========================================================================
 
-    depends_build-append            port:swig-octave
-    configure.args-strsed           s/--disable-swig-octave/--enable-swig-octave/
+subport ${name}-java {
+  categories-prepend    java
+  description           Java bindings for ${description}
+  long_description      ${long_description} This package provides Java \
+                        bindings, modules, and scripts.
 
-}
+  # Need GNU Classpath to get jni.h header
+  depends_build-append      bin:javac:gcc48 port:swig-java port:gnu-classpath
+  depends_build-append      port:swig-java
+  depends_lib-append        bin:java:kaffe
+  depends_lib-append        port:${name}
 
-variant swig_java description "Enable SWIG Java interface" {
+  configure.javac           /usr/bin/javac -source 1.5 -target 1.5
+  configure.args-replace    -DENABLE_SWIG_JAVA=no -DENABLE_SWIG_JAVA=yes
 
-    # Need GNU Classpath to get jni.h header
-    depends_build-append            bin:javac:gcc48 port:swig-java port:gnu-classpath
-    depends_lib-append              bin:java:kaffe
-    configure.args-strsed           s/--disable-swig-java/--enable-swig-java/
+  destroot.target           install
+  destroot.args-append      -C ${worksrcpath}/../build/swig/java
 
+  livecheck.type        none
 }
 
-variant sasl description "Use cyrus-sasl2 for authentication" conflicts gssapi {
+#========================================================================
+# Create subports for MATLAB
+#========================================================================
 
-    configure.args-replace          --without-sasl --with-sasl=${prefix}
-    depends_lib-append              port:cyrus-sasl2
+subport ${name}-matlab {
+  categories-prepend    matlab
+  description           MATLAB bindings for ${description}
+  long_description      ${long_description} This package provides MATLAB \
+                        bindings, modules, and scripts.
 
-}
+  depends_lib-append        port:${name}-java
 
-variant gssapi description "Use kerberos5 gssapi for authentication" conflicts sasl {
+  destroot.target           install
 
-    configure.args-replace          --without-gssapi --with-gssapi=${prefix}
-    depends_lib-append              port:kerberos5
+  configure.args-replace    -DENABLE_SWIG_JAVA=no -DENABLE_SWIG_JAVA=yes
```

Comment:
If this also uses swig-java, it should also declare a build dependency on it using `depends_build-append port:swig-java`.